src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/_params.py

```diff
@@ -213,6 +213,11 @@ def load_arguments(self, _):
         c.argument('os_type', get_enum_type(aci_connector_os_type),
                    help='The OS type of the connector')
 
+    with self.argument_context('aks update-credentials') as c:
+        c.argument('reset_service_principal', action='store_true', options_list=['--reset-service-principal', '-s'])
```

Comment:
TLDR: `-s` is okay.

`-s` is not a global shortcut. It is a common shortcut for subscription but because other commands used `-s` before the global `--subscription` was added (for example, in storage it is `-s` for share) we couldn't do such a breaking change. Use of `--subscription` wasn't deemed to be a frequent enough operation that giving it a short option was practical.

Reply:
It's not global but it's common enough that I think it's confusing to re-use for a different `az aks` option. But either way is fine, that's just my opinion.